### PR TITLE
Fix compose

### LINKED ISSUE #27: [Bug] Compose doesn't forward pytorch functions to internal transforms
**Description**
`Compose` uses a `list` to store the transformation which leads to some problems when specific functions of `torch.nn.Module` should also be applied to children (e.g. the `to()` method).

Quick fix: change `list` to `torch.nn.ModuleList` which limits our transformation to `rising` transforms which are subclasses of `torch.nn.Module`

Other solutions: look at which functions we really need and overwrite them appropriately (should at least fix problems with `to()`)

Status: looking for a better solution because that is not really satisfying... 
Any ideas @justusschock @haarburger ?

### PR DESCRIPTION
**Pull request by justusschock**
## Short Description
Changes the internal container of the compose class to `torch.nn.ModuleList` to fix #27 

### PR Implementer
This is a small checklist for the implementation details of this PR.
If you submit a PR, please look at these points (don't worry about the `RisingTeam`
and `Reviewer` workflows, the only purpose of those is to have a compact view of
the steps)

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/PhoenixDL/rising/blob/master/CONTRIBUTING.md).

- [x] Implementation
- [x] Docstrings & Typing
- [x] Check `__all__` sections and `__init__`
- [x] Unittests (look at the line coverage of your tests, the goal is 100%!)
- [x] Update notebooks & documentation if necessary
- [x] Pass all tests
- [x] Add the checksum of the last implementation commit to the Changelog

### RisingTeam
<details>
  <summary>RisingTeam workflow</summary>

  - [ ] Add pull request to project (optionally delete corresponding project note)
  - [ ] Assign correct label (if you don't have permission to do this, someone will do it for you. 
      Please make sure to communicate the current status of the pr.)
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)
</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>
  
  - [ ] Do all tests pass? (Unittests, NotebookTests, Documentation)
  - [ ] Does the implementation follow `rising` design conventions?
  - [ ] Are the tests useful? (!!!) Are additional tests needed? 
        Can you think of critical points which should be covered in an additional test?
  - [ ] Optional: Check coverage locally / Check tests locally if GPU is necessary to execute
</details>

### CODE REVIEW

File: rising/transforms/compose.py [hunk 1]
```diff
@@ -42,7 +68,12 @@ def __init__(self, *transforms,
         super().__init__(grad=True)
         if isinstance(transforms[0], Sequence):
             transforms = transforms[0]
-        self.transforms = transforms
+
+        for idx, trafo in enumerate(transforms):
+            if not isinstance(trafo, torch.nn.Module):
+                transforms[idx] = _TransformWrapper(trafo)
+
+        self.transforms = torch.nn.ModuleList(transforms)
```
Comment (mibaumgartner) [inline rising/transforms/compose.py]: [Optional] Should we do this with a property? (with a setter)

Comment (justusschock) [reply rising/transforms/compose.py]: I don't know, if this wouldn't be an overkill for this one time, we assign it, but I can definitely add a check, whether this already is a module list (but even if it is, this should be fine as calls would be propagated through)

Comment (mibaumgartner) [reply rising/transforms/compose.py]: Yes :) It was just optional ^^
